tests/virt_autotest/reboot_and_wait_up_upgrade.pm

```diff
@@ -48,4 +48,18 @@ sub run {
     $self->reboot_and_wait_up($timeout);
 }
 
+sub enable_y2log_analysis {
+    # Create and Upload y2log for analysis
+    assert_script_run "save_y2logs /tmp/y2logs.tar.bz2", 180;
+    upload_logs("/tmp/y2logs.tar.bz2");
+    save_screenshot;
+}
+
+sub post_fail_hook {
+    my ($self) = @_;
+    reset_consoles;
+    select_console('root-console');
```

Comment:
What is this console ? It is a ssh console or tty console ? Which one you intended to use ?

Reply:
There should be ssh console.

Reply:
I think the original reset_console better to be kept before select_console, because it increases the chance to have active(connected) console, especially during upgrade.

Reply:
> There should be ssh console.

I think you can help make it more robust a little bit. Try tty console if ssh console is not available. It is not hard to do, just call that console name. So if ( ! select_console(root-console)) { select_console(tty-console) }.

Reply:
> I think the original reset_console better to be kept before select_console, because it increases the chance to have active(connected) console, especially during upgrade.

agree on this, imho, during offline host upgrade failure (e.g timeout issue),  better to use reset_consoles to make sure
that *ALL* consoles are closed, then used with "ssh console" to call upload_y2logs

Reply:
> 
> I think you can help make it more robust a little bit ? Try tty console if ssh console is not available. It is not hard to do, just call that console name. So if ( ! select_console(root-console)) { select_console(tty-console) } (just pseudo code)

Good point, had double confirm just root-console (ssh console) work very well during upgrade failure without tty-console.
So, suggest to used with root-console (ssh-console) to call upload_y2logs